chain/network/src/types.rs

```diff
@@ -263,25 +263,48 @@ impl From<HandshakeV2> for Handshake {
     }
 }
 
+/// Contains metadata used for routing messages to particular `PeerId` or `AccountId`.
 #[cfg_attr(feature = "deepsize_feature", derive(DeepSizeOf))]
 #[derive(BorshSerialize, BorshDeserialize, PartialEq, Eq, Clone, Debug)]
-pub struct SyncData {
-    pub(crate) edges: Vec<Edge>,
-    pub(crate) accounts: Vec<AnnounceAccount>,
+pub struct RoutingTableSync {
+    /// List of known edges from `RoutingTableActor::edges_info`.
+    edges: Vec<Edge>,
+    /// List of known `account_id` to `PeerId` mappings.
+    /// Useful for `send_message_to_account` method, to route message to particular account.
+    accounts: Vec<AnnounceAccount>,
 }
 
-impl SyncData {
-    pub(crate) fn edge(edge: Edge) -> Self {
-        Self { edges: vec![edge], accounts: Vec::new() }
+impl RoutingTableSync {
+    pub(crate) fn new(edges: Vec<Edge>, accounts: Vec<AnnounceAccount>) -> Self {
+        Self { edges, accounts }
+    }
+
+    pub(crate) fn from_edges(edges: Vec<Edge>) -> Self {
+        Self { edges, accounts: Default::default() }
+    }
+
+    pub(crate) fn from_accounts(accounts: Vec<AnnounceAccount>) -> Self {
+        Self { edges: Default::default(), accounts }
     }
 
-    pub fn account(account: AnnounceAccount) -> Self {
+    /// Construct from a single edge.
+    pub(crate) fn from_edge(edge: Edge) -> Self {
+        Self { edges: vec![edge], accounts: Default::default() }
+    }
+
+    /// Construct from account.
+    pub fn from_account(account: AnnounceAccount) -> Self {
         Self { edges: Vec::new(), accounts: vec![account] }
     }
 
+    /// Is empty.
     pub(crate) fn is_empty(&self) -> bool {
         self.edges.is_empty() && self.accounts.is_empty()
     }
+
+    pub(crate) fn unpack(self) -> (Vec<Edge>, Vec<AnnounceAccount>) {
+        (self.edges, self.accounts)
+    }
```

Comment:
Not sure if encapsulating fields makes sense *in this case*. The struct is `derive(BorshSerialize)` so the fact that it is a pair of two vecs is exposed anyway.

Reply:
I make those fields `pub(crate)`